### PR TITLE
replace this package with https://github.com/beacon-biosignals/MsgPack2.jl

### PR DESCRIPTION
**Pull request by jrevels**
Let's see how CI runs correctly here; seems like codecov/Travis are already set up so should work in theory

### CODE REVIEW

**Comment by jrevels** [comment]
Awesome, looks all green. 

I'll wait for an approval here, then I'll merge and tag a v1.0.0 release (major version bump since it's quite a big code change, though I made sure to leave in deprecation paths where they seemed needed).

File: src/pack.jl [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+"""
+    pack(io::IO, x)
+
+Like `pack(x)`, but write the resulting bytes to `io`.
+
+See also: [`unpack`](@ref)
+"""
+function pack(io::IO, x)
+    pack_type(io, msgpack_type(typeof(x)), x)
+    return io
```
[the remark this answers is not in the document]
Comment (jrevels) [reply src/pack.jl]: `nothing` seems like a sensible return value here so I've gone ahead and made that change

File: src/pack.jl [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+# serialize does (e.g. an object field with a `Union` type). Thus, before giving
+# up, we first attempt to resolve the issue by calling `msgpack_type(typeof(x))`
+# directly.
+function pack_type(io, t::AnyType, x)
+    tx = msgpack_type(typeof(x))
+    if tx isa AnyType
+        error("no non-`AnyType` MsgPack mapping found for ", typeof(x), "; please ",
+              "overload `msgpack_type` for this type.")
+    end
+    return pack_type(io, tx, x)
```
[the remark this answers is not in the document]
Comment (jrevels) [reply src/pack.jl]: Yup, good point - the return values of these functions (which are just meant to be internal) should probably be `nothing`. Just would require a bit of churn to refactor; I'll open a follow-on issue.

[the remark this answers is not in the document]
Comment (kmsquire) [reply src/pack.jl]: That would be fine.  How about just creating our own write function?  Something like

```julia
function write(args...)
    Base.write(args...)
    nothing
end
```

(Probably simple enough that `@inline` isn't necessary?)

**Pull request merged**